pkg/kubelet/sysctl/allowlist_test.go

```diff
@@ -84,3 +91,60 @@ func TestAllowlist(t *testing.T) {
 		}
 	}
 }
+
+func TestAdmit(t *testing.T) {
+	tests := []struct {
+		attrs          *lifecycle.PodAdmitAttributes
+		expectedResult lifecycle.PodAdmitResult
+	}{
+		{
+			attrs: &lifecycle.PodAdmitAttributes{Pod: &metav1.Pod{}, OtherPods: []*metav1.Pod{}},
+			expectedResult: lifecycle.PodAdmitResult{
+				Admit: true,
+			},
+		},
+		{
+			attrs: &lifecycle.PodAdmitAttributes{Pod: &metav1.Pod{
+				Spec: metav1.PodSpec{
+					SecurityContext: &metav1.PodSecurityContext{
+						Sysctls: []metav1.Sysctl{
+							{Name: "kernel.shm_rmid_forced"},
+						},
+					},
+					HostIPC:     false,
+					HostNetwork: false,
+				},
+			}, OtherPods: []*metav1.Pod{}},
+			expectedResult: lifecycle.PodAdmitResult{
+				Admit: true,
+			},
+		},
+		{
+			attrs: &lifecycle.PodAdmitAttributes{Pod: &metav1.Pod{
+				Spec: metav1.PodSpec{
+					SecurityContext: &metav1.PodSecurityContext{
+						Sysctls: []metav1.Sysctl{
+							{Name: "kernel.shm_rmid_forced"},
+						},
+					},
+					HostIPC:     true,
+					HostNetwork: false,
+				},
+			}, OtherPods: []*metav1.Pod{}},
+			expectedResult: lifecycle.PodAdmitResult{
+				Admit:   false,
+				Reason:  ForbiddenReason,
+				Message: fmt.Sprint("forbidden sysctl: \"kernel.shm_rmid_forced\" not allowed with host ipc enabled"),
+			},
+		},
+	}
+	w, err := NewAllowlist(append(SafeSysctlAllowlist(), "kernel.msg*", "kernel.sem"))
+	if err != nil {
+		t.Fatalf("failed to create allowlist: %v", err)
+	}
+
+	for _, test := range tests {
+		result := w.Admit(test.attrs)
+		assert.Equal(t, test.expectedResult, result)
```

Comment:
```suggestion
		require.Equal(t, test.expectedResult, result)
```
prefer require over assert - it terminates the test rather than stacking errors in a way that can obscure things while debugging.